### PR TITLE
Backport docs: update and format

### PR DESCRIPTION
**Pull request by ramonjd**
## What?
Updating docs surrounding backports and PHP changes.

## Why?
The Gutenberg PHP doc was out of date given the backport change log process.

Also, I formatted and updated the examples for the backport change log process.

### CODE REVIEW

File: backport-changelog/readme.md [hunk 1]
```diff
@@ -1,16 +1,65 @@
...
+
+For individual PRs, there are two Github labels that can be used to exclude a PR from the backport changelog CI check:
+
+- `Backport from WordPress Core` - Indicates that the PR is a backport from WordPress Core and doesn't need a Core PR.
+- `No Core Sync Required` - Indicates that any changes do not need to be synced to WordPress Core.
+
+If there are specific file or directory changes that should **never** be flagged as requiring a Core backport PR, you can add it to the list of exceptions in [.github/workflows/check-backport-changelog.yml](https://github.com/WordPress/gutenberg/tree/trunk/.github/workflows/check-backport-changelog.yml).
+
+
+
```
Comment (andrewserong) [inline backport-changelog/readme.md]: Is this extra whitespace accidental?

File: lib/README.md [hunk 2]
```diff
@@ -192,12 +192,12 @@ Existing comments in `lib/load.php` should act as a guide.
...
+On open Gutenberg PRs, changes to certain files are flagged as requiring syncing (also called "backporting") to WordPress Core, for example, PHP files in `/lib` and PHP unit tests.
 
-The Gutenberg GitHub pull request in question should be labeled with the `Needs PHP backport` label if the changes are to be synced to Core.
+The CI checks will indicate whether you need to create a Core PR. If you do, you'll need to create a corresponding markdown file and place it within the appropriate release subdirectory in the [Core backport changelog](https://github.com/WordPress/gutenberg/tree/trunk/backport-changelog/).
 
-If so, it is recommended to create a [new Trac ticket](https://core.trac.wordpress.org/newticket) and submit a pull request to the [WordPress Core GitHub repository](https://github.com/WordPress/wordpress-develop) soon after your pull request is merged.
+For more information, please refer to the [Core backport changelog documentation](https://github.com/WordPress/gutenberg/tree/trunk/backport-changelog/readme.md).
```
Comment (andrewserong) [inline lib/README.md:199]: I like that this is linking to the other docs for the full set of instructions 👍

File: backport-changelog/readme.md [hunk 3]
```diff
@@ -1,16 +1,65 @@
...
+For example, if your Core PR number is `1234` and is slated to be part of the WordPress 6.9 release, the filename will be `1234.md`, and will be placed in the `/backport-changelog/6.9` directory.
+
+The content of the markdown file should be the Github URL of the Core PR, followed by a list of Gutenberg PR Github URLs whose changes are backported in the Core PR.
+
+A single Core PR may contain changes from one or multiple Gutenberg PRs.
+
+### Examples
+
+Let's say the next WordPress release is 6.9. You have two Gutenberg PRs  — `1111` and `2222` — whose changes are backported in a single Core PR, number `1234`.
```
Comment (andrewserong) [inline backport-changelog/readme.md]: Tiniest of whitespace nits:

```suggestion
Let's say the next WordPress release is 6.9. You have two Gutenberg PRs — `1111` and `2222` — whose changes are backported in a single Core PR, number `1234`.
```

**Review by andrewserong — COMMENTED** [review]
This looks like a good change to me! Just pinging @noisysocks and @kevin940726 for visibility since they'll be the next release leads for 6.7 🙂

File: backport-changelog/readme.md (hunk 1 above)
Comment (ramonjd) [reply backport-changelog/readme.md]: Accident. Updated. Thanks!

**Review by andrewserong — APPROVED** [review]
Whoops, I meant to hit the `Approve` button on my comment earlier, so here you go! ✅

**Comment by noisysocks** [comment]
Very nice thanks!

**Comment by ramonjd** [comment]
Thanks for checking, team!

**Comment by gziolo** [comment]
That’s extremely helpful, thank you for contributing these documentation 👏

**Pull request merged**